### PR TITLE
Create troubleshooting.md

### PR DESCRIPTION
**Pull request by flands**
**Documentation:** Create troubleshooting.md

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+## Common Issues
+
+### Collector restarts
+
+The Collector may restart if it is not
+[sized properly](https://github.com/open-telemetry/opentelemetry-collector/blob/master/docs/performance.md)
+or configured properly, for example a queue size configured higher than
+available memory. In most cases, restarts are due to memory
+pressure. To mitigate this, we recommend you pass the
+`--mem-ballast-size-mib` parameter to the `otelcol` process as well as
```
Comment (owais) [inline docs/troubleshooting.md]: This isn't really intended to solve this problem. It's mainly to improve the throughput by reducing GC cycles.

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+configure the [memory_limiter](https://github.com/open-telemetry/opentelemetry-collector/tree/master/processor#memory-limiter)
+processor. Alternatively, restarts may be due to resource limits
+configured on the Colleector.
+
+### Data being dropped
+
+Data may be dropped for a variety of reasons, but most commonly because of an:
+
+- [Improperly sized Collector](https://github.com/open-telemetry/opentelemetry-collector/blob/master/docs/performance.md))
```
Comment (tigrannajaryan) [inline docs/troubleshooting.md]:
```suggestion
- [Improperly sized Collector](https://github.com/open-telemetry/opentelemetry-collector/blob/master/docs/performance.md)) resulting in Collector being unable to process and export the data as fast as it is received.
```

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+- Exporter destination unavailable
```
Comment (tigrannajaryan) [inline docs/troubleshooting.md]:
```suggestion
- Exporter destination unavailable or accepting the data too slowly.
```

File: docs/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+- The client configuration is incorrect
+
+### Exporting data not working
+
+If you are unable to export to a destination then this is likely because
+either:
+
+- There is a network configuration issue
+- The exporter configuration is incorrect
+- The destination is available
```
Comment (pjanotti) [inline docs/troubleshooting.md]:
```suggestion
- The destination is unavailable
```

File: docs/troubleshooting.md (hunk 1 above)
Comment (flands) [reply docs/troubleshooting.md]: Good point, removing and recommending the memory limiter processor only

File: docs/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+extension, which by default is on port `55679`, can be used to check RPC
+stats via `/debug/rpcz` as well as trace data via `/debug/tracez`.
+
+- The [pprof](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#pprof)
+extension, which by default is on port `1777`, allows you to profile the
+Collector as it runs.
+
+## Common Issues
+
+### Collector restarts
```
[the remark this answers is not in the document]
Comment (flands) [reply docs/troubleshooting.md]: Good point, let me address

File: docs/troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+- [Local exporters](https://github.com/open-telemetry/opentelemetry-collector/tree/master/exporter#local-exporters)
+can be configured to inspect the data being processed by the Collector.
+
+- The [health_check](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#health_check)
+extension, which by default is on port `13133`, can be used to ensure
+the Collector is functioning properly.
+
+- The [zpages](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#zpages)
+extension, which by default is on port `55679`, can be used to check RPC
+stats via `/debug/rpcz` as well as trace data via `/debug/tracez`.
```
Comment (pjanotti) [inline docs/troubleshooting.md]: RPC stats via `/debug/rpcz` are available only for gRPC and I don't think they have much value for typical users, not sure if it is worth mentioning. Anyway, at least clarify that `rpcz` is available for "some" receivers/exporters.

File: docs/troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+The Collector may exit/restart if it is not
+[sized properly](https://github.com/open-telemetry/opentelemetry-collector/blob/master/docs/performance.md)
+or configured properly, for example a queue size configured higher than
+available memory. In most cases, restarts are due to memory
+pressure. To mitigate this, we recommend you configure the
+[memory_limiter](https://github.com/open-telemetry/opentelemetry-collector/tree/master/processor#memory-limiter)
+processor.
+
+Note: restarts may be due to resource limits configured on the Colleector.
```
Comment (pjanotti) [inline docs/troubleshooting.md]:
```suggestion
Note: restarts may be due to resource limits configured on the Collector.
```

File: docs/troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+- [Local exporters](https://github.com/open-telemetry/opentelemetry-collector/tree/master/exporter#local-exporters)
+can be configured to inspect the data being processed by the Collector.
+
+- The [health_check](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#health_check)
+extension, which by default is on port `13133`, can be used to ensure
+the Collector is functioning properly.
+
+- The [zpages](https://github.com/open-telemetry/opentelemetry-collector/blob/master/extension/README.md#zpages)
+extension, which by default is on port `55679`, can be used to check
+trace data via `/debug/tracez`.
```
Comment (pjanotti) [inline docs/troubleshooting.md]:
```suggestion
receivers and exporters operations via `/debug/tracez`.
```

**Pull request merged**